py-pkg/pkgnet/function_reporter.py

```diff
@@ -0,0 +1,88 @@
+import inspect
+
+import pandas as pd
+
+from pkgnet.abstract_graph_reporter import AbstractGraphReporter
+from pkgnet.graphs import DirectedGraph
+from pkgnet.package_report import register_reporter
+from pkgnet.search_utils import (
+    get_all_modules_in_package,
+    get_fully_qualified_name,
+    get_object,
+    get_package_name,
+    safe_import_module,
+)
+
+
+@register_reporter
+class FunctionReporter(AbstractGraphReporter):
+
+    _graph_class = DirectedGraph
+
+    report_template = "tab_function_report.jinja"
+    report_slug = "function-report"
+    report_name = "Functions"
+    layout = "kamada_kawai_layout"
+
+    ### PROPERTIES ###
+
+    ### PUBLIC METHODS ###
+
+    ### PRIVATE METHODS ###
+
+    def _extract_nodes_and_edges(self):
+        if self.pkg_name is None:
+            raise AttributeError("pkg_name is not set for this reporter.")
+
+        modules = get_all_modules_in_package(self.pkg_name)
+
+        # Get all functions used in modules
+        pkg_fcns = set()
+        for module_name in modules:
+            module_obj = safe_import_module(module_name)
+            if module_obj is None:
+                continue
+            pkg_fcns |= {
```

Comment:
`|=` wow cool!

Reply:
The Python devs have thought of everything. You can do `&=` for in-place intersection updates.